Fortinet/fortigate/ingest/parser.yml

```diff
@@ -0,0 +1,230 @@
+name: fortigate
+pipeline:
+  - name: parsed_event
+    external:
+      name: cef.parse-cef
+    filter: '{{original.message.startswith("CEF")}}'
+
+  - name: parsed_event
+    external:
+      name: kv.parse-kv
+      properties:
+        value_sep: '='
+        item_sep: ','
+    filter: "{{original.message | re_match('^\\s*\\w+=[^,]+(?:,\\w+=.*)\\s*$')}}"
+
+  - name: parsed_event
+    external:
+      name: kv.parse-kv
+      properties:
+        value_sep: '='
+        item_sep: \s
+    filter: "{{original.message | re_match('^\\s*\\w+=[^\\s]+(?:\\s\\w+=.*)\\s*$')}}"
+
+  - name: field_extraction
+
+stages:
+  field_extraction:
+    actions:
+      - set:
+          action.name: "{{parsed_event.message.name or parsed_event.message.FTNTFGTaction or parsed_event.message.FortinetFortiGateaction or parsed_event.message.act or parsed_event.message.action}}"
+          action.outcome: "{{parsed_event.message.status or 'success'}}"
+          action.outcome_reason: "{{parsed_event.message.outcome_reason or parsed_event.message.msg.split('\n')[0].strip('\"')}}"
+          action.type: "{{parsed_event.message.subtype or parsed_event.message.FTNTFGTsubtype or parsed_event.message.FortinetFortiGatesubtype}}"
+          destination.address: "{{parsed_event.message.dstip or parsed_event.message.dst}}"
+          destination.bytes: "{{parsed_event.message.rcvdbyte or parsed_event.message.out}}"
+          destination.domain: "{{parsed_event.message.hostname}}"
+          destination.ip: "{{parsed_event.message.dstip or parsed_event.message.dst}}"
+          destination.mac: "{{parsed_event.message.dstmac}}"
+          destination.nat.ip: "{{parsed_event.message.tranip or parsed_event.message.destinationTranslatedAddress}}"
+          destination.nat.port: "{{parsed_event.message.destinationTranslatedPort}}"
+          destination.packets: "{{parsed_event.message.rcvdpkt or parsed_event.message.FTNTFGTrcvpkt or parsed_event.message.FortinetFortiGatercvdpkt or parsed_event.message.get('Packets Received')}}"
+          destination.port: "{{parsed_event.message.dstport or parsed_event.message.dpt}}"
+          dns.question.name: "{{parsed_event.message.qname or parsed_event.message.FTNFGTqname or parsed_event.message.FortinetFortiGateqname}}"
+          dns.question.type: "{{parsed_event.message.qtype or parsed_event.message.FTNFGTqtype or parsed_event.message.FortinetFortiGateqtype}}"
+          event.action: "{{parsed_event.message.reason}}"
+          event.category: "{{parsed_event.message.type}}"
+          event.code: "{{parsed_event.message.logid or parsed_event.message.FTNTFGTlogid or parsed_event.message.FortinetFortiGatelogid}}"
+          event.reason: "{{parsed_event.message.msg}}"
+          event.severity: "{{parsed_event.message.severity}}"
+          event.type: "{{parsed_event.message.desc}}"
```

Comment:
`event.type` has a restricted [list of value](https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-type.html). Another field should be used